app/restore_to.go

```diff
@@ -0,0 +1,137 @@
+package app
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/backupstore"
+	"github.com/rancher/longhorn-engine/util"
+	"github.com/urfave/cli"
+)
+
+const (
+	DefaultImageFormat = "qcow2"
+	QEMUImageBinary    = "qemu-img"
+)
+
+var SupportedImageFormats = []string{
+	"qcow2",
+	"raw",
+}
+
+func RestoreToCmd() cli.Command {
+	return cli.Command{
+		Name: "restore-to",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "backing-file",
+				Usage: "filepath or dirpath containing exactly one qcow2 backing file",
+			},
+			cli.StringFlag{
+				Name:  "backup-url",
+				Usage: "backup URL to be published",
+			},
+			cli.StringFlag{
+				Name:  "image-format",
+				Usage: "format of image to produce",
+				Value: DefaultImageFormat,
+			},
+			cli.StringFlag{
+				Name:  "output-file",
+				Usage: "filepath to write the resulting image to",
+				Value: "volume.qcow2",
+			},
+		},
+		Action: func(c *cli.Context) {
+			if err := restoreTo(c); err != nil {
+				logrus.Fatalf("Error running restore to file command: %v", err)
+			}
+		},
+	}
+}
+
+func imageFormatSupported(desiredFormat string) bool {
+	for _, supportedFormat := range SupportedImageFormats {
+		if desiredFormat == supportedFormat {
+			return true
+		}
+	}
+	return false
+}
+
+func restoreTo(c *cli.Context) error {
+	imageFormat := c.String("image-format")
+	if !imageFormatSupported(imageFormat) {
+		return fmt.Errorf("unsupported image format: %s", imageFormat)
+	}
+
+	backingFile, err := openBackingFile(c.String("backing-file"))
+	if err != nil {
+		return err
+	}
+
+	backupURL := c.String("backup-url")
+	if backupURL == "" {
+		return fmt.Errorf("backup-url must be provided")
+	}
+
+	backupFilepath := "backup.qcow2"
+
+	if err := backupstore.RestoreDeltaBlockBackup(backupURL, backupFilepath); err != nil {
+		return err
+	}
+
+	outputFile := c.String("output-file")
+	if backingFile == nil {
```

Comment:
`backingFile` wasn't used? Seems you only use the path.

Reply:
It became unnecessary after the last code review changes, I forgot to remove it.